File: src/main/java/com/kinisoftware/upcomingMovies/HelpIntentHandler.java
package com.kinisoftware.upcomingMovies;

import java.util.Optional;

import static com.amazon.ask.request.Predicates.intentName;
import static com.kinisoftware.upcomingMovies.UpcomingMoviesStreamHandler.CARD_TITLE;

import com.amazon.ask.dispatcher.request.handler.HandlerInput;
import com.amazon.ask.dispatcher.request.handler.RequestHandler;
import com.amazon.ask.model.Response;

public class HelpIntentHandler implements RequestHandler {

	@Override
	public boolean canHandle(HandlerInput input) {
		return input.matches(intentName("AMAZON.HelpIntent"));
	}

	@Override
	public Optional<Response> handle(HandlerInput input) {
		String text = "Pregúntame  por los estrenos de cine de esta semana";
		return input.getResponseBuilder()
				.withSpeech(text)
				.withSimpleCard(CARD_TITLE, text)
				.withReprompt(text)
				.build();
	}
}

File: src/main/kotlin/com/kinisoftware/upcomingMovies/handler/HelpIntentHandler.kt
package com.kinisoftware.upcomingMovies.handler

import com.amazon.ask.dispatcher.request.handler.HandlerInput
import com.amazon.ask.dispatcher.request.handler.RequestHandler
import com.amazon.ask.model.Response
import com.amazon.ask.request.Predicates.intentName
import com.kinisoftware.upcomingMovies.Translations
import com.kinisoftware.upcomingMovies.getLanguage
import java.util.Optional

class HelpIntentHandler : RequestHandler {

    override fun canHandle(input: HandlerInput): Boolean {
        return input.matches(intentName("AMAZON.HelpIntent"))
    }

    override fun handle(input: HandlerInput): Optional<Response> {
        val text = Translations.getMessage(input.getLanguage(), Translations.TranslationKey.HELP)
        return input.responseBuilder
                .withSpeech(text)
                .withReprompt(text)
                .build()
    }
}
